### DISCUSSION TITLE
Adding 404 page to a custom path

### DISCUSSION THREAD

**Discussion by uilianries** [community/community#23792] [Other Feature Feedback, Questions, & Ideas]
<p>Hi!</p>
<p>I would like to use a custom 404.html which is generated by Sphinx, but the file is not inserted on root level.</p>
<p>To be more specific, this file <a href="https://github.com/conan-io/docs/blob/gh-pages/en/latest/404.html" rel="nofollow noopener">https://github.com/conan-io/docs/blob/gh-pages/en/latest/404.html</a> should be used for default 404 error.</p>
<p>However, it does not, so I’ve added a default 404.html to redict: <a href="https://github.com/conan-io/docs/blob/gh-pages/404.html" rel="nofollow noopener">https://github.com/conan-io/docs/blob/gh-pages/404.html</a></p>
<p>Is there some way to do it, without that file on the root level? (I’m not using Jekyll).</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi, <a class="mention" href="/uilianries">@uilianries</a>!</p>
<p>I’m afraid GitHub Pages only supports storing a custom 404 page in the root of your publishing source.</p>
<p>That said, rather than creating a unique HTML file at this location, and within it a JavaScript redirect, you could instead consider using a symlink.</p>
<p>By creating a symlink to <code>en/latest/404.html</code>, you can save yourself the hassle and inelegance of the redirect. Instead, GitHub Pages will simply follow the symlink during the build process, and implement your desired custom 404 page.</p>
<p>On macOS, Linux, or Git Bash, you can create the necessary symlink by running this command from the root of your repository’s <code>gh-pages</code> branch:</p>
<pre><code class="lang-auto">ln -s en/latest/404.html
</code></pre>
<p>I hope that helps!</p>
<p>Friday</p>

**Comment by uilianries** [comment]
<p>Thanks a lot <a class="mention" href="/darrowby385">@darrowby385</a>!</p>
